### PR TITLE
AStar/Datastructures update

### PR DESCRIPTION
**Pull request by Markolie**
Changes:
1) Updates A* to /tg/'s latest. Updates by Menshin: https://github.com/tgstation/-tg-station/pull/12076. 
Hopefully this'll make A* a little bit less laggy.
2) Adds Datastructures (in case we ever need them, currently unused). By RemieRichards: https://github.com/tgstation/-tg-station/pull/11396.
3) Fixes grammar in human examine texts. Fixes https://github.com/ParadiseSS13/Paradise/issues/2240.
4) Adds base simple_animals to the blacklist. Fixes https://github.com/ParadiseSS13/Paradise/issues/2223.

Do not merge until #12076 on /tg/ is merged.

### CODE REVIEW

**Comment by tigercat2000** [comment]
What... even are the datastructures for?

File: code/game/machinery/bots/farmbot.dm [hunk 1]
```diff
@@ -1,597 +0,0 @@
...
-#define FARMBOT_MODE_FERTILIZE	 	2
-#define FARMBOT_MODE_WEED			3
-#define FARMBOT_MODE_REFILL			4
-#define FARMBOT_MODE_WAITING		5
-
-#define FARMBOT_ANIMATION_TIME		25 //How long it takes to use one of the action animations
-#define FARMBOT_EMAG_DELAY			60 //How long of a delay after doing one of the emagged attack actions
-#define FARMBOT_ACTION_DELAY		35 //How long of a delay after doing one of the normal actions
-
-/obj/machinery/bot/farmbot
```
Comment (tigercat2000) [inline code/game/machinery/bots/farmbot.dm]: RIP

**Comment by Markolie** [comment]
Wikipedia to the rescue: https://en.wikipedia.org/wiki/Data_structure

Basically it's a very efficient way of storing and retrieving data.

**Comment by optimumtact** [comment]
They don't have a use yet, but no harm in having them for future uses

**Comment by Fox-McCloud** [comment]
@Markolie 

You may want to take a look at A* in general and also some of the bots--I know TG updated both of those (primarily because of A*) to help address it consuming large portions of the CPU (which is a problem we currently have).

I think Aran updated it a while back.

**Comment by Markolie** [comment]
This is a full update of AStar, and our implementation for bots is identical.

**Comment by Fox-McCloud** [comment]
Alright; I believe that there's some vampire code (for bat spawning of all things) that uses Astar(), changeling sting too, though I dunno if either of those technically needs replaced or not.

File: code/game/machinery/bots/farmbot.dm (hunk 1 above)
Comment (FalseIncarnate) [inline code/game/machinery/bots/farmbot.dm]: I'd love to see farmbot make a comeback... but at the same time I don't want to mess with fixing farmbot.

File: code/game/machinery/bots/farmbot.dm [hunk 2]
```diff
@@ -1,597 +0,0 @@
...
-	var/setting_ignoreWeeds = 1
-	var/setting_ignoreMushrooms = 1
-
-	var/atom/target //Current target, can be a human, a hydroponics tray, or a sink
-	var/mode //Which mode is being used, 0 means it is looking for work
-
-	var/obj/structure/reagent_dispensers/watertank/tank // the water tank that was used to make it, remains inside the bot.
-
-	var/path[] = new() // used for pathing
-	var/frustration
```
Comment (Fox-McCloud) [inline code/game/machinery/bots/farmbot.dm]: farmbots gonna get all frustrated and fertilize the SHIT out of you....then collect it...to fertilize so more...

File: code/game/machinery/bots/farmbot.dm (hunk 1 above)
Comment (Fox-McCloud) [inline code/game/machinery/bots/farmbot.dm]: at one point in time farmbot did sorta kinda, halfway work---then Neca re-arranged botany and wrecked their pathing and they were confused.

File: code/defines/procs/AStar.dm [hunk 3]
```diff
@@ -237,9 +186,55 @@ proc/AStar(start,end,adjacent,dist,maxnodes,maxnodedepth = 30,mintargetdist,minn
...
+	if(DirBlockedWithAccess(src, adir, ID))
+		return 1
+	if(DirBlockedWithAccess(T, rdir, ID))
+		return 1
+	for(var/obj/O in T)
+		if(O.density && !istype(O, /obj/machinery/door) && !(O.flags & ON_BORDER))
+			return 1
+	return 0
+
+/proc/DirBlockedWithAccess(turf/T, dir, ID)
```
Comment (Menshin) [inline code/defines/procs/AStar.dm]: You're doubling the same proc that's already present in the unsorted.dm (l. 208) file in your codebase.
Same with ```LinkBlocked``` and ```LinkBlockedWithAccess```.

**Comment by Markolie** [comment]
Can be merged.

**Pull request merged**